src/zenkit/core/vec4f.py

```diff
@@ -0,0 +1,235 @@
+import math
+from ctypes import Structure, c_float
+from typing import ClassVar, Any
+
+class Vec4f(Structure):
+    """
+    A class to represent a 4D vector using floats, supporting various vector operations.
+    
+    Attributes:
+        x (float): The x-component of the vector.
+        y (float): The y-component of the vector.
+        z (float): The z-component of the vector.
+        w (float): The w-component of the vector.
+    """
+
+    _fields_: ClassVar[tuple[str, Any]] = [
+        ("x", c_float),
+        ("y", c_float),
+        ("z", c_float),
+        ("w", c_float),
+    ]
+
+    def __repr__(self) -> str:
+        """
+        Return a string representation of the vector.
+
+        Returns:
+            str: A string in the format 'Vec4f(x, y, z, w)'.
+        """
+        return f"Vec4f(x={self.x}, y={self.y}, z={self.z}, w={self.w})"
+    
+    def __neg__(self):
+        """
+        Negate the vector (reverse the direction of the vector).
+
+        Returns:
+            Vec4f: A new Vec4f instance with negated components.
+        """
+        return Vec4f(-self.x, -self.y, -self.z, -self.w)
+    
+    def __eq__(self, other):
+        """
+        Check if two vectors are equal component-wise.
+
+        Args:
+            other (Vec4f): The vector to compare with this vector.
+
+        Returns:
+            bool: True if the vectors are equal, False otherwise.
+
+        Raises:
+            TypeError: If the comparison is not between two Vec4f objects.
+        """
+        if isinstance(other, Vec4f):
+            return self.x == other.x and self.y == other.y and self.z == other.z and self.w == other.w
+        raise TypeError("Comparison is only supported between two Vec4f objects")
```

Comment:
`__eq__` should return `False` instead of throwing when comparing against an object of another type.